extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetailsPage.ts

```diff
@@ -61,13 +75,116 @@ export class AssessmentDetailsPage extends MigrationWizardPage {
 	}
 
 	public async onPageEnter(pageChangeInfo: azdata.window.WizardPageChangeInfo): Promise<void> {
-		await this._header.populateAssessmentDetailsHeader();
-		await this._body.populateAssessmentBody();
+		this.wizard.registerNavigationValidator((pageChangeInfo) => {
+			this.wizard.message = { text: '' };
+			if (pageChangeInfo.newPage < pageChangeInfo.lastPage) {
+				return true;
+			}
+			this.executeChange(this.migrationStateModel._targetType);
+			const errors: string[] = [];
+			if (this.migrationStateModel._databasesForMigration.length === 0) {
+				errors.push(constants.SELECT_DATABASE_TO_MIGRATE);
+			}
+			if (errors.length > 0) {
+				this.wizard.message = {
+					text: errors.join(EOL),
+					level: azdata.window.MessageLevel.Error
+				};
+				return false;
+			}
+			return true;
+		});
+		this.migrationStateModel._targetType = MigrationTargetType.SQLDB;
+		this.executeChange(this.migrationStateModel._targetType);
+		await this._header.populateAssessmentDetailsHeader(this.migrationStateModel);
+		await this._body.populateAssessmentBody(this.migrationStateModel);
+		this.wizard.nextButton.enabled = this.migrationStateModel._assessmentResults !== undefined;
 	}
 
 	public async onPageLeave(pageChangeInfo: azdata.window.WizardPageChangeInfo): Promise<void> {
+		this.wizard.registerNavigationValidator(pageChangeInfo => true);
+		this.wizard.message = { text: '' };
 	}
 
 	protected async handleStateChange(e: StateChangeEvent): Promise<void> {
 	}
+
+	private didUpdateDatabasesForMigration(priorDbs: string[], selectedDbs: string[]) {
+		this.migrationStateModel._didUpdateDatabasesForMigration = selectedDbs.length === 0
+			|| selectedDbs.length !== priorDbs.length
+			|| priorDbs.some(db => selectedDbs.indexOf(db) < 0);
+	}
+
+	// function to give target type based on value selected in dropdown.
+	private getTargetTypeBasedOnSelection(targetType: string): MigrationTargetType {
+		switch (targetType) {
+			case constants.SUMMARY_SQLDB_TYPE:
+				return MigrationTargetType.SQLDB;
+			case constants.SUMMARY_VM_TYPE:
+				return MigrationTargetType.SQLVM;
+			case constants.SUMMARY_MI_TYPE:
+				return MigrationTargetType.SQLMI;
+		}
+		return MigrationTargetType.SQLDB;
```

Comment:
why default type is DB here.

Reply:
When we come at assessment result page, by default a target type is selected, user can change that target type. Based on discussion with Abhishek, we can select any one target type as default. For entire page when a user jumps to target selection and assessment result page, I have chosen SQLDB as default selection. We can discuss to change default selection for user if needed.

Reply:
In earlier experience, by default nothing was selected. If Mock-up asks this, i am Okey.